sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/options/ShareCreateOptions.java

```diff
@@ -102,4 +103,29 @@ public ShareCreateOptions setRootSquash(ShareRootSquash rootSquash) {
         this.rootSquash = rootSquash;
         return this;
     }
+
+    /**
+     * Get the enableSnapshotVirtualDirectoryAccess property: The EnableSnapshotVirtualDirectoryAccess property.
+     * Optional. Supported in version 2023-08-03 and above.
+     * Specifies whether the snapshot virtual directory should be accessible at the root of share mount point when NFS is enabled.
+     * If not specified, the default is true.
+     * @return the enableSnapshotVirtualDirectoryAccess value.
+     */
+    public Boolean isEnableSnapshotVirtualDirectoryAccess() {
+        return enableSnapshotVirtualDirectoryAccess;
+    }
+
+    /**
+     * Set the enableSnapshotVirtualDirectoryAccess property:
+     * Optional. Supported in version 2023-08-03 and above.
+     * Specifies whether the snapshot virtual directory should be accessible at the root of share mount point when NFS is enabled.
+     * If not specified, the default is true.
+     * @param enableSnapshotVirtualDirectoryAccess the enableSnapshotVirtualDirectoryAccess value to set.
+     * @return the ShareCreateOptions object itself.
+     */
+    public ShareCreateOptions setEnableSnapshotVirtualDirectoryAccess(
```

Comment:
`setSnapshotVirtualDirectoryAccessEnabled(Boolean snapshotVirtualDirectoryAccessEnabled)`

Same for the other options class